cep-??.md

```diff
@@ -0,0 +1,134 @@
+<table>
+<tr><td> Title </td><td> Specification of <code>environment.yml</code> input files </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> June 4, 2024 </td></tr>
+<tr><td> Updated </td><td> June 4, 2024 </td></tr>
+<tr><td> Discussion </td><td> https://github.com/conda/ceps/pull/81 </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CEP standardizes the format of `environment.yml` input files for conda clients.
+
+## Motivation
+
+The motivation of this CEP is merely informative. It describes the details of an existing file format.
+
+## Nomenclature
+
+The `environment.yml` file format was introduced by `conda env`. It has been traditionally referred to as "the YAML input file", but other YAML files exist in the ecosystem now. To avoid confusion, in this CEP we will only use the term `environment.yml`. Do note that the actual filename of an input file using this format might be a different one (e.g. `env.yml`).
+
+## Specification
+
+`environment.yml` files are YAML documents that encode the necessary information to create a new conda environment. They can contain up to five top-level keys:
+
+- `name`: The preferred name for the environment.
+- `prefix`: The preferred full path for the environment. Often ignored.
+- `dependencies`: List of package specifications that MUST be installed in the new environment.
+- `channels`: conda channels that will be used to solve the dependencies
+- `variables`: Environment variables that SHOULD be added to the `conda-meta/state` file in the resulting environment.
+
+Additional sections can be present, but they MUST be ignored and the user SHOULD receive an informative warning about them.
+
+The file name is not standardized, but the extension MUST be `yml` or `yaml`.
+
+### `name` and `prefix`
+
+Optional, `str`.
+
+Both fields refer to the _preferred_ name or path for the newly created environment. Tools SHOULD allow these suggestions to be overridden by the user with additional CLI flags or equivalent. If the proposed environment path exists, tools MUST NOT overwrite silently by default.
+
+Special names `base` and `root` SHOULD not be accepted. Prefixes targetting protected system paths SHOULD be rejected. Paths can contain tildes (`~`) and environment variables, and they MUST be expanded when present.
+
+The name of the environment (and the last component of the expanded`prefix` path) MUST NOT contain any of these characters: `/`, ` `, `:`, `#`.
+
+### `dependencies`
+
+Required, `list[str | dict[str, Any]]`.
+
+The simplest form for this section is a list of `str` encoding `MatchSpec`-compatible requirements. These items MUST be processed as conda requirements and submitted to the solver (along with `channels`) to obtain the contents for the resulting conda environment.
```

Comment:
Should platform selectors be mentioned here or is that implied with `MatchSpec`-compatible?

Reply:
AFAIK, `environment.yml` doesn't support platform selectors like `meta.yaml`, doesn't it?